inter-ident-verif/README.md

```diff
@@ -0,0 +1,62 @@
+___
+## Proposed solution
+Many companies have various branches that operate different parts of the company. For example, if Apple Tech Support were to call you, it likely wouldn't be Tim Cook on the other end. Instead, it would be some branch of Apple that put on a tightly vetted list of Apple-trusted callers. If a caller calls a user and the user needs to verify that branch is a trusted Apple caller, they need a way to verify this with only the knowledge that the caller is supposedly an Apple-verified source. This is where Kanidm comes into play.
+
+All users (individuals, companies, company branches, etc.) of this verification system will have an existing RSA key linked to their Kanidm credentials on the servers. This key will not change (unless for very good reason), and the private key is only known by the exact account without exception. Since these keys are linked to credentials, we can safely find the authentic public key of any user we want, without real time communication between servers around the world.
+
+Say Apple Tech Support is calling a user. They claim to work for Apple, and the user wants to verify. The user doesn't care about the exact identity of the caller, only that they are Apple-trusted, since no scam company could make Apple's tightly vetted list of trusted branches. Since they're not contacting Apple directly, there are three key connections that must be verified:
+1. The user needs to know the caller is Apple-trusted.
+2. Apple needs to verify the caller is trusted.
+3. The user must verify their request for verification was directly verified by Apple and nobody else.
+
+This can be accomplished by sending an encrypted message to Apple, and then sending it back to the user with a different encryption pattern. If the message received is the same as the message sent, then all identities are verified. These guarentees can be made with the following messaging pattern:
+
+### User
+1. User generates random, unique message, and stores a copy for later.
+2. User encrypts message with their private key, ensuring that the end target (Apple) must acknowledge them when modifying the result.
+3. User encrypts result with Apple public key, preventing the caller from decrypting to get the original message.
+4. User sends result to Apple Tech Support.
+
+### Apple Tech Support
+5. Apple Tech Support encrypts message with Apple Tech Support private key, ensuring that Apple knows it's from them.
+6. Apple Tech Support sends result to Apple.
+
+### Apple
```

Comment:
It's worth defining all the parties involved in this operation, and where kanidm sits in this. At the moment I can only assume "Apple" is the kanidm site.

It's probably also worth not saying Apple but "corp".